orvara/orbit.pyx

```diff
@@ -1075,29 +1111,52 @@ def calc_RV(Data data, Params par, Model model):
+                                one_d_240, pi, pi_d_2, tanEAd2, sqrt1pe_div_sqrt1me, RVampl,
+                                cosarg, sinarg, ecccosarg, fabs(model.sinEA[i]), fabs(model.EA[i]))
+    # calculate the relative rv's
+    cdef int i_rel_RV = data.nTot - data.n_rel_RV
+    cdef double conv = -1. * (
+                par.msec + par.mpri) / par.msec  # conversion factor from RV of the primary to delta RV = RVsecondary - RVprimary.
+    for i in range(data.n_rel_RV):
+        j = i + i_rel_RV
+        if iplanet == data.rel_RV_planetID[i]:
+            model.rel_RV[i] = _calc_RV(model.sinEA[j], model.cosEA[j], model.EA[j], one_d_24,
+                                       one_d_240, pi, pi_d_2, tanEAd2, sqrt1pe_div_sqrt1me, RVampl * conv,
+                                       cosarg, sinarg, ecccosarg, fabs(model.sinEA[j]), fabs(model.EA[j]))
+    # Don't use the following: we do about 20 times better above.
+    #for i in range(data.nRV):
+    #    TA = 2*atan2(sqrt1pe*sin(model.EA[i]/2), sqrt1me*cos(model.EA[i]/2))
+    #    model.RV[i] += RVampl*(cos(TA + par.arg) + par.ecc*cos(par.arg))
+    return
 
-        if fabs(model.sinEA[i]) > 1.5e-2:
-            tanEAd2 = (1 - model.cosEA[i])/model.sinEA[i]
-        elif model.EA[i] < -pi or model.EA[i] > pi:
-            raise ValueError("EA input to calc_RV must be betwen -pi and pi.")
-        elif fabs(model.EA[i]) < pi_d_2:
-            EA = model.EA[i]
-            tanEAd2 = EA*(0.5 + EA**2*(one_d_24 + one_d_240*EA**2))
-        elif model.sinEA[i] != 0:
-            tanEAd2 = (1 - model.cosEA[i])/model.sinEA[i]
-        else:
-            tanEAd2 = 1e100
-
-        ratio = sqrt1pe_div_sqrt1me*tanEAd2
-        fac = 2/(1 + ratio**2)
-        model.RV[i] += RVampl*(cosarg*(fac - 1) - sinarg*ratio*fac + ecccosarg)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.nonecheck(False)
+cdef _calc_RV(double sinEA, double cosEA, double EA, double one_d_24, double one_d_240,
+              double pi, double pi_d_2, double tanEAd2, double sqrt1pe_div_sqrt1me, double RVampl,
+              double cosarg, double sinarg, double ecccosarg, double abs_sinEA, double abs_EA):
+
+    if abs_sinEA > 1.5e-2:
+        tanEAd2 = (1 - cosEA) / sinEA
+    elif EA < -pi or EA > pi:
+        raise ValueError("EA input to calc_RV must be between -pi and pi.")
+    elif abs_EA < pi_d_2:
+        tanEAd2 = EA * (0.5 + EA ** 2 * (one_d_24 + one_d_240 * EA ** 2))
+    elif sinEA != 0:
+        tanEAd2 = (1 - cosEA) / sinEA
+    else:
+        tanEAd2 = 1e100
 
+    ratio = sqrt1pe_div_sqrt1me * tanEAd2
+    fac = 2 / (1 + ratio ** 2)
+    return RVampl * (cosarg * (fac - 1) - sinarg * ratio * fac + ecccosarg)
     # Don't use the following: we do about 20 times better above.
     #for i in range(data.nRV):
     #    TA = 2*atan2(sqrt1pe*sin(model.EA[i]/2), sqrt1me*cos(model.EA[i]/2))
     #    model.RV[i] += RVampl*(cos(TA + par.arg) + par.ecc*cos(par.arg))
 
-    return
```

Comment:
@t-brandt I wanted to confirm that this lone "return" was extraneous. It is in orvara/master currently as well. As far as I can tell, this has no functionality.

Reply:
Confirmed.